src/compiler/transformers/declarations.ts

```diff
@@ -0,0 +1,1276 @@
+                        node.heritageClauses = createNodeArray(map(node.heritageClauses, clause => {
+                            if (clause.token === SyntaxKind.ExtendsKeyword) {
+                                const oldDiag = getSymbolAccessibilityDiagnostic;
+                                getSymbolAccessibilityDiagnostic = declarations.createGetSymbolAccessibilityDiagnosticForNode(clause.types[0]);
+                                const newClause = updateHeritageClause(clause, map(clause.types, t => updateExpressionWithTypeArguments(t, visitNodes(t.typeArguments, visitDeclarationComponents), newId)));
+                                getSymbolAccessibilityDiagnostic = oldDiag;
+                                return newClause;
+                            }
+                            return updateHeritageClause(clause, visitNodes(createNodeArray(filter(clause.types, t => isEntityNameExpression(t.expression) || t.expression.kind === SyntaxKind.NullKeyword)), visitDeclarationComponents));
+                        }));
+                        return [statement, cleanup(node)];
+                    }
+                    else {
+                        node.heritageClauses = transformHeritageClauses(input.heritageClauses);
+                    }
+                    return cleanup(node);
+                }
+                case SyntaxKind.VariableStatement: {
+                    if (!forEach(input.declarationList.declarations, getBindingNameVisible)) return;
+                    const nodes = visitNodes(input.declarationList.declarations, visitDeclarationComponents);
+                    if (!length(nodes)) return;
+                    const node = getMutableClone(input);
+                    node.decorators = undefined;
+                    node.modifiers = createNodeArray(ensureModifiers(input));
+                    node.declarationList = updateVariableDeclarationList(input.declarationList, nodes);
+                    return cleanup(node);
+                }
+                case SyntaxKind.EnumDeclaration: {
+                    const node = getMutableClone(input);
+                    node.decorators = undefined;
+                    node.modifiers = createNodeArray(ensureModifiers(input));
+                    node.members = createNodeArray(mapDefined(input.members, m => {
+                        if (shouldStripInternal(m)) return;
+                        // Rewrite enum values to their constants, if available
+                        const constValue = resolver.getConstantValue(m);
+                        return preserveJsDoc(updateEnumMember(m, m.name, constValue !== undefined ? createLiteral(constValue) : undefined), m);
+                    }));
+                    return cleanup(node);
+                }
+            }
+
+            Debug.assertNever(input, `Unhandled top-level node in declaration emit: ${(ts as any).SyntaxKind[(input as any).kind]}`);

```

Comment:
Yeah, that's a bit of a recent style update, and this PR's been in progress for months and in review for more than a month; it could stand to be updated (also cc @andy-ms we should consider enforcing it with a lint rule, actually) :P